3-enrich/emr-etl-runner/lib/snowplow-emr-etl-runner/s3_tasks.rb

```diff
@@ -37,9 +51,17 @@ module S3Tasks
       #
       # Parameters:
       # +region+:: the region to add into the filenames
-      Contract String => Func[String, String => String]
-      def self.build_fix_filenames(region)
+      Contract String, String => Func[String, String => String]
+      def self.build_fix_filenames(region, collector_format)
         return lambda { |basename, filepath|
+
+          if matches = collector_format.match(/^.+?\/(.+?)\//)
+            adapter = matches.captures[0]
+            if adapter == "com.urbanairship.connect"
+              return generate_short_ua_filename(filepath)
+            end
+          end
+
```

Comment:
Nice - you made it very easy to visually verify that the existing behaviour is safely unchanged @ninjabear!